junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ScriptExecutionCondition.java

```diff
@@ -98,57 +103,53 @@ private String createSource(String[] lines) {
 		return String.join(System.lineSeparator(), lines);
 	}
 
-	private Worker getScriptExecutionWorker(ExtensionContext context) {
-		ExtensionContext root = context.getRoot();
-		ExtensionContext.Store rootStore = root.getStore(NAMESPACE);
-		return rootStore.getOrComputeIfAbsent(WORKER, this::createWorker, Worker.class);
-	}
-
-	// Create worker via reflection to hide the `javax.script` dependency.
-	private Worker createWorker(String name) {
-		logger.debug(() -> String.format("Creating instance of `%s`...", name));
-		Optional<Throwable> optionalThrowable = assumeScriptEngineIsLoadableByClassForName();
-		if (optionalThrowable.isPresent()) {
-			return new ThrowingWorker(optionalThrowable.get());
-		}
-		try {
-			return (Worker) Class.forName(name).getDeclaredConstructor().newInstance();
-		}
-		catch (ReflectiveOperationException cause) {
-			logger.error(cause, () -> String.format("Creating instance of `%s` failed", name));
-			return new ThrowingWorker(cause);
-		}
-	}
-
-	// This method may fail on JREs that don't provide the "javax.script" package at all
-	// or it fails on JREs launched with an active module system using insufficient module
-	// graphs, i.e. the application does not read "java.scripting" module.
-	Optional<Throwable> assumeScriptEngineIsLoadableByClassForName() {
-		try {
-			Class.forName("javax.script.ScriptEngine");
-			return Optional.empty();
-		}
-		catch (Throwable cause) {
-			logger.error(cause, () -> "Loading `javax.script.ScriptEngine` failed");
-			return Optional.of(cause);
-		}
-	}
-
 	/**
 	 * Evaluates scripts and returns a conditional evaluation result.
 	 */
-	interface Worker {
-		ConditionEvaluationResult work(ExtensionContext context, List<Script> scripts);
+	interface Evaluator {
+
+		ConditionEvaluationResult evaluate(ExtensionContext context, List<Script> scripts);
+
+		/**
+		 * Create evaluator via reflection to hide the `javax.script` dependency.
+		 */
+		static Evaluator forName(String name) {
+			Optional<Throwable> optionalThrowable = assumeScriptEngineIsLoadableByClassForName();
+			if (optionalThrowable.isPresent()) {
+				return new ThrowingEvaluator(optionalThrowable.get());
+			}
+			try {
+				return (Evaluator) Class.forName(name).getDeclaredConstructor().newInstance();
+			}
+			catch (ReflectiveOperationException cause) {
+				return new ThrowingEvaluator(cause);
+			}
+		}
+
+		/**
+		 * This method may fail on JREs that don't provide the "javax.script" package at all
+		 * or it fails on JREs launched with an active module system using insufficient module
+		 * graphs, i.e. the application does not read "java.scripting" module.
+		 */
+		static Optional<Throwable> assumeScriptEngineIsLoadableByClassForName() {
```

Comment:
maybe change the method name to `...ViaReflection` instead of `ByClassForName`...